src/Nonlinear/ReverseAD/Coloring/Coloring.jl

```diff
@@ -0,0 +1,547 @@
+module Coloring
+
+import DataStructures
+
+include("topological_sort.jl")
+
+# indexed sparse set of integers
+mutable struct IndexedSet
+    nzidx::Vector{Int}
+    empty::BitArray{1}
```

Comment:
```suggestion
    empty::BitVector
```
We usually use Vector and Matrix aliases?

Reply:
Done.